packages/yoroi-ergo-connector/src/inject.js

```diff
@@ -108,6 +126,24 @@ class ErgoAPI {
 
 const ergo = Object.freeze(new ErgoAPI());
 `
+
+const cardanoApiInject = `
+
+// @todo add cardano rpc resolver when it is done
+
```

Comment:
Do we not want to reuse the RPC relay instead? They're both going to the same extension and for RPCs should be basically the same and afaik there's no naming conflicts (we can always add in a type: ' cardano' | 'ergo'  too which could be useful regardless).

Although this would also depend how we're handling it on the Yoroi side. There should be a LOT of things in common between the two, and most of the code here + the RPC stuff on both sides is pretty independent from the actual API that's being implemented across the RPCs.

It's worth noting that having the RPC code being injected/existing != you are connected. You could have been disconnected at some point anyway so the RPC already handles the case where you no longer have permissions to access the API so that shouldn't be an issue reusing it. It's all async so it shouldn't block interaction anywhere between the two.

Reply:
Yes, this is definetely what we would want to achieve, either right away from the beginning or as a reworking at a later stage. Thank you, for the details, @rooooooooob ! 🙌